### PR TITLE
[image_picker_android] Removes unneeded permissions from tests

### PR DESCRIPTION
**Pull request by tarrinneal**
Some tests were using a check for permissions that is not needed for image_picker to function. This pr removes them.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by stuartmorgan — COMMENTED** [review]
Thanks, i noticed this while I was working on `MethodCall` untangling yesterday and was thinking we should clean it up. Can you update the test names too so they don't say that's part of the test state?

File: packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java [hunk 1]
```diff
@@ -148,8 +148,7 @@ public void chooseMultiImageFromGallery_whenPendingResultExists_finishesWithAlre
 
   @Test
   @Config(sdk = 30)
-  public void
-      chooseImageFromGallery_whenHasExternalStoragePermission_launchesChooseFromGalleryIntent() {
+  public void chooseImageFromGallery_launchesChooseFromGalleryIntent() {
     when(mockPermissionManager.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE))
```
[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java]: so I did...

**Review by stuartmorgan — APPROVED** [review]
LGTM

File: packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java [hunk 2]
```diff
@@ -148,10 +148,7 @@ public void chooseMultiImageFromGallery_whenPendingResultExists_finishesWithAlre
 
   @Test
   @Config(sdk = 30)
-  public void
-      chooseImageFromGallery_whenHasExternalStoragePermission_launchesChooseFromGalleryIntent() {
-    when(mockPermissionManager.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE))
-        .thenReturn(true);
+  public void chooseImageFromGallery_LaunchesChooseFromGalleryIntent() {
```
[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java:151]: for sure, I had some weird merge conflict issues I should have handled more carefully.

[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/image_picker/image_picker_android/android/src/test/java/io/flutter/plugins/imagepicker/ImagePickerDelegateTest.java]: found a couple you missed too I think

**Pull request merged**